cpp/oneapi/dal/algo/svm/backend/gpu/misc.hpp

```diff
@@ -0,0 +1,124 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include "oneapi/dal/backend/primitives/ndarray.hpp"
+#include "oneapi/dal/backend/primitives/sort/sort.hpp"
+
+namespace oneapi::dal::svm::backend {
+
+namespace pr = dal::backend::primitives;
+
+enum class ws_edge { up, low };
+
+inline sycl::event arrange(sycl::queue& queue,
+                           pr::ndview<std::uint32_t, 1>& indices_sort,
```

Comment:
Why does it work with `std::uint32_t` instead of `std::int32_t` and `std::int64_t`?